src/main/java/org/openstreetmap/atlas/checks/validation/tag/LongNameCheck.java

```diff
@@ -0,0 +1,111 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import static org.openstreetmap.atlas.checks.constants.CommonConstants.COLON;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.walker.OsmWayWalker;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.tags.names.NameFinder;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Flags features with name greater than a configurable number of characters.
+ *
+ * @author bbreithaupt
+ */
+public class LongNameCheck extends BaseCheck<Long>
+{
+    private static final long serialVersionUID = -8395117392888327533L;
+    private static final long NAME_MAX_DEFAULT = 40;
+    private static final Set<String> NAME_TAG_KEYS = NameFinder.STANDARD_TAGS_NON_REFERENCE.stream()
+            .map(Validators::findTagNameIn).collect(Collectors.toSet());
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections.singletonList(
+            "Feature {0,number,#} has the following tags with over {1,number,#} characters: {2}.");
+
+    private final long nameMax;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public LongNameCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.nameMax = this.configurationValue(configuration, "name.max", NAME_MAX_DEFAULT);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return !this.isFlagged(object.getOsmIdentifier())
```

Comment:
Since the check looks at all AtlasObjects, if two atlas objects of different types, say relation and way has long names, wouldn't marking the osmIdentifier instead of atlasIdentifier , skip one of the objects?

Reply:
Good catch! The atlas identifier has the same issue, but we can create a unique osm id.

Reply:
oh yeah! Thanks!